File: backend/models/attendance.py
from __future__ import annotations

from datetime import datetime


def build_student_entry(student_id: str, student_name: str, present: bool = False, marked_at=None):
    return {
        "student_id": student_id,
        "student_name": student_name,
        "present": present,
        "marked_at": marked_at,
    }


def build_session_document(payload: dict, students: list[dict]):
    return {
        "date": payload.get("date"),
        "subject": payload.get("subject"),
        "department": payload.get("department"),
        "year": payload.get("year"),
        "division": payload.get("division"),
        "created_at": datetime.utcnow(),
        "finalized": False,
        "ended_at": None,
        "students": students,
    }

File: frontend/app/models/attendance.ts
export interface MatchResult {
  user_id: string;
  name: string;
}

export interface RecognitionFace {
  box: [number, number, number, number];
  match: MatchResult | null;
  confidence?: number | null;
  status?: string;
  message?: string;
  already_marked?: boolean;
}

export interface AttendanceRecord {
  studentId: string;
  studentName: string;
  date: string;
  subject: string;
  department: string;
  year: string;
  division: string;
  status: "present" | "absent";
  markedAt?: string | null;
}

export interface AttendanceStats {
  totalStudents: number;
  presentToday: number;
  absentToday: number;
  attendanceRate: number;
}

export interface SessionHistoryItem {
  sessionId: string;
  date: string;
  subject: string;
  department: string;
  year: string;
  division: string;
  finalized: boolean;
  presentCount: number;
  studentCount: number;
}
